cmake/dependencies.cmake

```diff
@@ -26,22 +26,29 @@ fetch_models_and_validation_set()
 get_linux_name(LINUX_OS_NAME)
 
 if(CMAKE_CROSSCOMPILING AND CMAKE_HOST_SYSTEM_NAME MATCHES Linux AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
-    set(protoc_version "3.18.2")
-
-    RESOLVE_DEPENDENCY(SYSTEM_PROTOC_ROOT
-        ARCHIVE_LIN "protoc-${protoc_version}-linux-x86_64.tar.gz"
-        TARGET_PATH "${TEMP}/protoc-${protoc_version}-linux-x86_64"
-        SHA256 "42fde2b6044c1f74c7e86d4e03b43aac87128ddf57ac6ed8c4eab7a1e21bbf21"
-    )
-    debug_message(STATUS "host protoc-${protoc_version} root path = " ${SYSTEM_PROTOC_ROOT})
-
-    reset_deps_cache(SYSTEM_PROTOC)
-
-    find_host_program(
-        SYSTEM_PROTOC
-        NAMES protoc
-        PATHS "${SYSTEM_PROTOC_ROOT}/bin"
-        NO_DEFAULT_PATH)
+    if (NOT LINUX_OS_NAME STREQUAL "CHROMIUMOS")
+        set(protoc_version "3.18.2")
+
+	RESOLVE_DEPENDENCY(SYSTEM_PROTOC_ROOT
+		ARCHIVE_LIN "protoc-${protoc_version}-linux-x86_64.tar.gz"
+		TARGET_PATH "${TEMP}/protoc-${protoc_version}-linux-x86_64"
+		SHA256 "42fde2b6044c1f74c7e86d4e03b43aac87128ddf57ac6ed8c4eab7a1e21bbf21")
+	debug_message(STATUS "host protoc-${protoc_version} root path = " ${SYSTEM_PROTOC_ROOT})
+
+        reset_deps_cache(SYSTEM_PROTOC)
+
+        find_host_program(
+            SYSTEM_PROTOC
+            NAMES protoc
+            PATHS "${SYSTEM_PROTOC_ROOT}/bin"
+            NO_DEFAULT_PATH)
+    else()
+        find_host_program(
+            SYSTEM_PROTOC
+            NAMES protoc
+            PATHS "${SYSTEM_PROTOC_ROOT}/bin")
```

Comment:
who defines `SYSTEM_PROTOC_ROOT` in this case?

Reply:
chromeos sdk provides an older version of protoc library.  However if the newer version is required, goal is to pass the location of package (SYSTEM_PROTOC_ROOT) as cmake variable using ebuilds (https://chromium.googlesource.com/chromiumos/docs/+/HEAD/portage/ebuild_faq.md#ebuild-faq)

If protoc is only used by front end libraries (Paddle / Tf / Onnx), can we download only if front end libraries are enabled like here.. 
https://github.com/openvinotoolkit/openvino/blob/9dee25fa79e771c7a4a5759c6021bb2af687d292/thirdparty/CMakeLists.txt#L85